src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private static final String SEPARATOR = "\n";
+    private static final String PREFIX = "fruit,quantity\n";
+    private static final String SUFFIX = "\n";
+    private static final String DELIMITER = ",";
+
+    @Override
+    public String getReport() {
+        return Storage.getAllFruits().entrySet().stream()

```

Comment:
Make sure that the Storage.getAllFruits() method is properly encapsulated. If it's returning a direct reference to the storage map, it could lead to potential issues with data integrity.